sw/device/tests/crypto/kdf_hmac_ctr_functest.c

```diff
@@ -0,0 +1,1115 @@
+  };
+  uint32_t km_data[] = {
+      0x1fa14960, 0xa71ae98b, 0x750a55d4, 0xac1325a0,
+      0xbeb88709, 0xd66134ba, 0xf61959e8, 0x4091784c,
+  };
+
+  kdf_test_vector_t test = {
+      .key_mode = kOtcryptoKeyModeHmacSha384,
+      .key_derivation_key = kdk_data,
+      .kdk_bytelen = 32,
+      .kdf_context = context_data,
+      .kdf_context_bytelen = sizeof(context_data),
+      .kdf_label = label_data,
+      .kdf_label_bytelen = sizeof(label_data),
+      .km_mode = kOtcryptoKeyModeAesCbc,
+      .keying_material = km_data,
+      .km_bytelen = 32,
+  };
+  return run_test(&test);
+}
+
+/**
+ * Test case 8:
+ *
+ * Basic test case with HMAC SHA384 using 384 bit input and output keys
+ *
+ * KDF Mode = HMAC Counter
+ * KDK      = 0x0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b (48 octets)
+ * context  = (0 octets)
+ * label    = (0 octets)
+ * L        = 384 bits
+ *
+ * KM       = 0xdfeb8693d58307a31d467e72de28dfc2
+ *              2af5d6610d2d25a409a517e412505936
+ *              b3d730de63521cfed045d50b5a047415 (48 octets)
+ */
+static status_t kdf_hmac_ctr_sha384_kdk48_km48_test(void) {
+  uint32_t kdk_data[] = {
+      0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b,
+      0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b, 0x0b0b0b0b,
+  };
+  uint32_t km_data[] = {
+      0x9386ebdf, 0xa30783d5, 0x727e461d, 0xc2df28de, 0x61d6f52a, 0xa4252d0d,
+      0xe417a509, 0x36595012, 0xde30d7b3, 0xfe1c5263, 0x0bd545d0, 0x1574045a,
+  };
+
+  kdf_test_vector_t test = {
+      .key_mode = kOtcryptoKeyModeHmacSha384,
+      .key_derivation_key = kdk_data,
+      .kdk_bytelen = 48,
+      .kdf_context = NULL,
+      .kdf_context_bytelen = 0,
+      .kdf_label = NULL,
+      .kdf_label_bytelen = 0,
+      .km_mode = kOtcryptoKeyModeAesEcb,
+      .keying_material = km_data,
+      .km_bytelen = 48,
+  };
+  return run_test(&test);
+}
+
+/**
+ * Test case 9:
+ *
+ * Basic test case with HMAC SHA384 using 2048 bit input and output keys
+ *
+ * KDF Mode = HMAC Counter
+ * KDK      = 0x0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
+ *              0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b (256 octets)
+ * context  = (8 octets)
+ * label    = (6 octets)
```

Comment:
I think some data is missing here, and for context/label in a few other test cases.

Reply:
I added the missing context and label comments.
Regarding the functional tests, since there are no test vectors in the specification, as in the case of HKDF, I used sw/device/tests/crypto/hkdf_functest.c as a template and generated keying material with a python script (pycryptodome library).